docs/api/fleet/agent_api_checkin.asciidoc

```diff
@@ -0,0 +1,65 @@
+[[fleet-agent-checkin]]
+=== Fleet agent checkin API
```

Comment:
Are the API names frozen, or can they still be changed? This name seems potentially confusing for developer audiences. They might think the API is related to versioning because there are two meanings:  

- The developer checked in his source code.  
- I checked in with the receptionist.

Aside from the actual endpoint name, I noticed that the Kibana APIs all start with verbs. Here are some possible titles you can consider:

Report Fleet agent state API (titleabbrev would be Report agent state)
Post Fleet agent state API (titleabbrev would be Post agent state)
Update Fleet agent state API (titleabbrev would be Update agent state)

@KOTungseth You have more recent experience writing about APIs than I have. What would you suggest?

Reply:
I changed a few of the API names, so I assume they are not frozen. My vote would be `Update Fleet agent state API`.

Reply:
This API is not a common one to use, It would only be used by the creator of a piece of software that would connect directly to fleet and thus would be controlled by Fleet.

Its worth noting that we used checkin because it has both those meanings @dedemorton. This API serves the role of Agents updating fleet as to their state, but it also has agents getting the latest configuration, and commands from fleet as to what it needs to do

Reply:
@mattapperson Does the stack have any other APIs that do something similar? I just don't expect users to think of "check in" when they're looking for an API that does what you describe. Anyhow, if you decide to keep the name, make short the short title follows are convention of starting with a verb.

Reply:
No nothing else in the stack (Besides BeatsCM) does this. Arguably, this IS fleet